wolfssl/wolfcrypt/types.h

```diff
@@ -784,7 +784,7 @@ decouple library dependencies with standard string, memory and so on.
 
     /* hash types */
     enum wc_HashType {
-    #if defined(HAVE_SELFTEST) || defined(HAVE_FIPS)
+    #if defined(HAVE_SELFTEST) || (defined(HAVE_FIPS) && (defined(HAVE_FIPS_VERSION) && (HAVE_FIPS_VERSION<=2)))
```

Comment:
In FIPSv1 (HAVE_FIPS), HAVE_FIPS_VERSION is not defined at all.  This change causes the unit test to fail for me using FIPSv1 and "--enable-fips --enable-opensslextra".

Was it possible you meant the following?

```
#if defined (HAVE_SELFTEST) || (defined(HAVE_FIPS) && (!defined(HAVE_FIPS_VERSION) || (HAVE_FIPS_VERSION < 2)))
```